Content.Server/GameObjects/Components/Explosion/ExplosiveComponent.cs

```diff
@@ -0,0 +1,125 @@
+        [Dependency] private readonly IMapManager _mapManager;
+        [Dependency] private readonly IServerEntityManager _serverEntityManager;
+        [Dependency] private readonly IEntitySystemManager _entitySystemManager;
+#pragma warning restore 649
+
+        public override string Name => "Explosive";
+
+        public int DamageMax = 1;
+        public int DamageFalloff = 1;
+        public int RangeDamageMax = 1;
+        public int Range = 1;
+        public int Delay = 1;
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+
+            serializer.DataField(ref DamageMax, "damageMax", 1);
+            serializer.DataField(ref DamageFalloff, "damageFalloff", 1);
+            serializer.DataField(ref RangeDamageMax, "rangeDamageMax", 1);
+            serializer.DataField(ref Range, "range", 1);
+        }
+
+        bool ITimerTrigger.Trigger(TimerTriggerEventArgs eventArgs)
+        {
+            var location = eventArgs.Source.Transform.GridPosition;
+
+            var entitiesAll = _serverEntityManager.GetEntitiesInRange(location, Range);
+
+            List<IEntity> entities = entitiesAll.ToList();
+
+            foreach (var entity in entitiesAll)
+            {
+                var distance = (int)entity.Transform.GridPosition.Distance(_mapManager, location);
+                if (!entity.Transform.IsMapTransform || distance > RangeDamageMax)
+                    continue;
+                if (entity.TryGetComponent(out DamageableComponent damagecomponent))
+                {
+                    damagecomponent.TakeDamage(DamageType.Brute, DamageMax);
+                }
+            }
+
+            foreach (var entity in entities)
+            {
+                var distance = (int)entity.Transform.GridPosition.Distance(_mapManager, location);
+                if (!entity.Transform.IsMapTransform || distance <= RangeDamageMax)
+                    continue;
+
+                if (entity.TryGetComponent(out DamageableComponent damagecomponent))
+                {
+                    
+                    var overallDamageFalloff = DamageFalloff;
+                    if (overallDamageFalloff > DamageMax) {
+                        overallDamageFalloff = DamageMax;
+                    }
+                    damagecomponent.TakeDamage(DamageType.Brute, DamageMax - distance*overallDamageFalloff);
+                }
+            }
+
+            var mapGrid = _mapManager.GetGrid(location.GridID);
+            //var circle = new Circle(location.Position, Range);
+            var aabb = new Box2(location.Position - new Vector2(Range / 2, Range / 2), location.Position + new Vector2(Range / 2, Range / 2));
+            var tiles = mapGrid.GetTilesIntersecting(aabb);
+            foreach (var tile in tiles)
+            {
+                var tileLoc = new GridCoordinates(tile.X, tile.Y, tile.GridIndex);
```

Comment:
```suggestion
                var tileLoc = mapGrid.GridTileToLocal(tile.GridIndices);
```

Also just so you know, your math was off for two reasons:

1. Tile indices do not map 1:1 to grid coordinates. Tiles can have any length.
2. You were getting coordinates at the bottom left corner of the tiles, which I assume wasn't intended.